### PR TITLE
IRSA-2685_sofia_add_example_element_to_naifid_resolver

### PR DESCRIPTION
**Pull request by deannaji**
Adding examples to NAIFID component, along with other enhancements.

Jira: https://jira.ipac.caltech.edu/browse/IRSA-2685
k8s: https://irsawebdev9.ipac.caltech.edu/irsa-2685/applications/sofia/
IFE-PR: https://github.com/IPAC-SW/irsa-ife/pull/64

### CODE REVIEW

File: src/firefly/js/ui/NaifidPanel.jsx [hunk 1]
```diff
@@ -127,14 +138,20 @@ NaifidPanelView.propTypes = {
 };
 
 
+function valueChanged(payload,fireValueChange){
+    let showHelp = false;
+    if(payload.value ==='') showHelp = true;
+    fireValueChange({message:payload.message, valid:payload.valid, displayValue:payload.value, showHelp:showHelp});
```
[the remark this answers is not in the document]
Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: Thanks, I completely didn't notice it!

File: src/firefly/js/ui/NaifidPanel.jsx [hunk 2]
```diff
@@ -127,14 +138,20 @@ NaifidPanelView.propTypes = {
...
+
 
 function updateFeedback(resolvedStr, valid, fireValueChange){
    const {objName, naifId}=  parseNaifPair(resolvedStr);
    const payload={
        feedback: resolvedStr,
        valid,
        displayValue: objName,
-       value: naifId,
+       value: objName+';'+naifId,//this is the returned value from the component.
```
[the remark this answers is not in the document]
Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: Completed, thanks!

**Comment by tgoldina** [comment]
Hi Dean, I thought more about it. Since you only set the value when you click on a suggestion, the only validation that makes sense is checking that the display value is matching the value set by suggestion. 
So I suggest storing the objName whenever the value changes, getting rid of the activeValidator (you can pass defValidator to SuggestBoxView), and doing validation in handleValueChange.

So your getSuggestions will look something like this:
```
    getSuggestions(val= '') {
            if (val.length<1) return [];
            const rval = resolveNaifidObj(val);
            if (!rval.p) return [];

            //if value has been searched previously, no need to call the api again.
            if(searchHistory.length > 0){
                const cachedSuggList = Object.entries(searchHistory).find(([k,v])=>(v.searchVal === val));
                if(cachedSuggList){
                    return Object.entries(cachedSuggList.pop().searchRes).map( ([k,v]) => `Object Name:${v.naifName}, NAIF ID:${v.naifId}`);
                }
            }

            return rval.p.then((response)=>{
                if(response.valid) {
                    const suggestionsList = Object.entries(response.data).map(([k,v])=>({naifId:v, naifName:k}));

                    searchHistory.push({'searchVal':val, 'searchRes': suggestionsList});
                    return Object.entries(suggestionsList).map( ([k,v]) => `Object Name:${v.naifName}, NAIF ID:${v.naifId}`);

                } else {
                    //console.error(response);
                    this.props.fireValueChange({valid: false, message: response.feedback});
                }
            });
    }
```
and handleValueChange like this:
```
function handleValueChange(payload, fireValueChange) {
    const newPayload= {...payload};
    newPayload.valid  = Boolean(newPayload.value) || (newPayload.displayValue === naifNamevalue);
    if (!newPayload.valid && !newPayload.message) {
        if (!newPayload.message) {
            newPayload.message = 'Please use name from the list';
        }
        if (naifNamevalue) {
            newPayload.value = '';
            newPayload.feedback = '';
        }
    }

    fireValueChange(newPayload) ;
}
```
`naifNamevalue` is set to objName whenever you click on suggestion (setting value into the field).

Notice that if the naif id was not resolved, the feedback makes into the validation message.

File: src/firefly/js/ui/NaifidPanel.jsx [hunk 3]
```diff
@@ -182,18 +161,23 @@ function renderSuggestion(str){
 
 function handleValueChange(payload, fireValueChange) {
     const newPayload= {...payload};
-    if (!payload.valid) {
-        newPayload.value= '';
-        newPayload.feedback= '';
+    newPayload.valid  = Boolean(newPayload.value) || (newPayload.displayValue === '' || newPayload.displayValue === naifNamevalue);
+    if (!newPayload.valid && !newPayload.message) {
+        if (!newPayload.message) {
```
Comment (tgoldina) [inline src/firefly/js/ui/NaifidPanel.jsx]: should be 
```
    if (!newPayload.valid) {
        if (!newPayload.message) {
```

Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: Done thanks!

Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: Done thanks!

File: src/firefly/js/ui/NaifidPanel.jsx [hunk 4]
```diff
@@ -34,53 +36,34 @@ class NaifidPanelView extends PureComponent {
...
+        if(searchHistory.length > 0){
+            const cachedSuggList = Object.entries(searchHistory).find(([k,v])=>(v.searchVal === val));
+            if(cachedSuggList){
+                return Object.entries(cachedSuggList.pop().searchRes).map( ([k,v]) => `Object Name:${v.naifName}, NAIF ID:${v.naifId}`);
+            }
+        }
+        return rval.p.then((response)=>{
+            if(response.valid) {
+                const suggestionsList = Object.entries(response.data).map(([k,v])=>({naifId:v, naifName:k}));
+                searchHistory.push({'searchVal':val, 'searchRes': suggestionsList});
```
[the remark this answers is not in the document]
Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: Done thanks!

File: src/firefly/js/ui/NaifidPanel.jsx (hunk 3 above)
Comment (deannaji) [reply src/firefly/js/ui/NaifidPanel.jsx]: For suggestions list sorting, this will be in a new ticket.

**Pull request merged**